sequence-diagrams/host-provisioning-xPU-VF.md

```diff
@@ -47,54 +48,73 @@ as a Kubernetes deployment.
+- Netdev CRD (New)
+- Pods
+- Service: reflecting the services in the tenant cluster.
+- EndpointSlice: reflecting the endpoint slices in the tenant cluster.
+- Network policy: reflecting the network policies in the tenant cluster.
 
-The following high level diagram provides an overview of the entities working
-together to provision a secondary network interface for a pod.
+### Multi Cluster sequence diagrams
 
-TODO
+#### Initialization
 
-#### Advantages
+The following high level diagram provides an overview of the initialization of the entities involved in provisioning an xPU VF to a host.
 
-The main advantage of this approach is that the scope of what is shared across clusters can be limited (in the two cluster scenario), while also allowing us to maintain a relatively simple CNI on the host side.
+![host-provisioning-initialization](./images/host-provisioning-xPU-VF-init.png)
 
-#### Required CRDs
+An alternative option would be to eliminate the host agent, and the sequence diagram for that looks as follows:
 
-TODO (WIP)
+![host-provisioning-initialization-no-host-agent](./images/host-provisioning-xPU-VF-init-no-host-agent.png)
 
-- Netdev (New)
-- Service: reflecting the services in the tenant cluster.
-- EndpointSlice: reflecting the endpoint slices in the tenant cluster.
-- Network policy
+#### Pod Creation
 
-### Single Cluster
+![host-provisioning-pod-creation](./images/host-provisioning-xPU-VF-pod-creation.png)
 
-The following sequence diagram provides an overview of the entities working
-together to provision an interface for a pod in the single cluster scenario.
+An alternative option would be to eliminate the host agent, and the sequence diagram for that looks as follows:
 
-![host-provisioning-xPU-VF](./images/host-provisioning-xPU-VF.png)
+![host-provisioning-pod-creation-no-host-agent](./images/host-provisioning-xPU-VF-pod-creation-no-host-agent.png)
 
-### Multi Cluster
+#### Pod Deletion
 
-The following sequence diagram provides an overview of the entities working
-together to provision an interface for a pod in the Multi cluster scenario.
+![host-provisioning-pod-deletion](./images/host-provisioning-xPU-VF-pod-deletion.png)
 
-![host-provisioning-xPU-VF-multi-cluster](./images/host-provisioning-xPU-VF-multi-cluster.png)
+An alternative option would be to eliminate the host agent, and the sequence diagram for that looks as follows:
 
-## CRDs
+![host-provisioning-pod-deletion-no-host-agent](./images/host-provisioning-xPU-VF-pod-deletion-no-host-agent.png)
 
-## DPUNetworkNodeState
+## CRDs
 
-```yaml
-TODO WIP
+### Netdev
+
+```golang
+type NetdevStatus string
+
+const (
+    Connected NetdevStatus = "connected"
+    Connecting NetdevStatus = "connecting"
+    Disconnected NetdevStatus = "disconnected"
+    ConnectionError NetdevStatus = "error"
+)
+
+type NetdevSpec struct {
+    NetdevID string `json:"netdev_id"` // SerialNum_PFID_SFID
```

Comment:
Purely for reference, this is what is passed in annotations in OVN-Kubernetes (written as a map indexed by the namespace):
```
type DPUConnectionDetails struct {
	PfId         string `json:"pfId"`
	VfId         string `json:"vfId"`
	SandboxId    string `json:"sandboxId"`
	VfNetdevName string `json:"vfNetdevName,omitempty"`
}
```


The IP info was already in the Pod Annotation, so it is not part of the DPU specific content.
```
// Internal struct used to marshal PodAnnotation to the pod annotation
type podAnnotation struct {
	IPs      []string   `json:"ip_addresses"`
	MAC      string     `json:"mac_address"`
	Gateways []string   `json:"gateway_ips,omitempty"`
	Routes   []podRoute `json:"routes,omitempty"`

	IP      string `json:"ip_address,omitempty"`
	Gateway string `json:"gateway_ip,omitempty"`
}

// Internal struct used to marshal PodRoute to the pod annotation
type podRoute struct {
	Dest    string `json:"dest"`
	NextHop string `json:"nextHop"`
}
```

Reply:
I forgot to mention, for IP, if I recall, `ip_address` and `gateway_ip` were there originally, and then when dual stack was added, it became an array of address, `ip_addresses` and `gateway_ips`. The original was left around for backwards compatibility.